pkg/azuredisk/nodeserver_v2.go

```diff
@@ -55,6 +58,45 @@ type NodeProvisioner interface {
 	GetBlockSizeBytes(devicePath string) (int64, error)
 }
 
+const (
+	detachInProcess    int32 = 0
+	detachNotInProcess int32 = 1
+)
+
+type deviceCheckerEntry struct {
+	diskURI         string
+	startTimestamp  time.Time
+	nextRetry       time.Time
+	detachInProcess int32
+}
+
+type deviceChecker struct {
+	lock  sync.RWMutex
+	entry *deviceCheckerEntry
+}
+
+// TODO: assess whether backoff is necessary or not.
+// In case of persisting failure, we will back off next recovery retry
+func (d *deviceChecker) backoff(diskURI string, lockAcquired bool) {
+	if !lockAcquired {
+		d.lock.Lock()
+		defer d.lock.Unlock()
+	}
+	if diskURI == d.entry.diskURI {
+		d.entry.nextRetry.Add(consts.DefaultMountRecoveryBackoff + d.entry.nextRetry.Sub(d.entry.startTimestamp)*consts.DefaultMountRecoveryBackoffScale)
```

Comment:
Since the base backoff delay is 2 minutes, an exponential backoff during a long running outage could result in the backoff delay in the hours if not days range. I imagine that Kubernetes will cease retrying well-before then, but we could end up delaying recovery by quite a bit after the outage ends. A fixed delay of 2 minutes is probably OK, but I still wonder if we should backoff at all.

Reply:
The intent of exponential backoff is to retry frequently at the beginning (under the assumption that the error may have been transient, short duration or one-off and the subsequent retry would succeed), but extend the backoff time if more retries are necessary to prevent overwhelming the server with request. The initial time is 2 minutes, so increasing the backoff doesn't really help anything. I believe we could get away with a constant duration backoff, or dropping the backoff entirely.